src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -949,31 +947,18 @@ private void WriteRedType(TreeType node)
                         Write($"public {OverrideOrNewModifier(field)}{GetRedPropertyType(field)} {field.Name}");
                         if (IsOptional(field))
                         {
-                            WriteLine();
-                            OpenBlock();
-                            WriteLine("get");
-                            OpenBlock();
-                            WriteLine($"var slot = ((Syntax.InternalSyntax.{node.Name})this.Green).{CamelCase(field.Name)};");
-                            WriteLine($"return slot != null ? new SyntaxToken(this, slot, {GetChildPosition(i)}, {GetChildIndex(i)}) : default;");
-                            CloseBlock();
-                            CloseBlock();
+                            WriteLine($" => ((InternalSyntax.{node.Name})this.Green).{CamelCase(field.Name)} is {{ }} slot ? new(this, slot, {GetChildPosition(i)}, {GetChildIndex(i)}) : default;");
```

Comment:
note: if feel like this could be generated to use a helper (like how we have `GetRed(ref this.field, fieldIndex))`).  We'd have a similar: `GetOptionalGreen(this, node, i)` and so on.  but that could very much change perf characteristics, and i didn't want to do that in this PR.  this PR is purely stylistic.  Making it so that the code we generate is much easier to work with when stepping in/out of hte generated file.